### PR TITLE
Refactor and add telnetlib

### PR DESCRIPTION
**Pull request by uSpike**

### CODE REVIEW

File: src/telnetio/_machine.py [hunk 1]
```diff
@@ -26,245 +22,263 @@ class TelnetOption:
...
+    #: IAC with multiple (3) bytes
+    negotiation = auto()
+    #: Subnegotation option
+    subn_option = auto()
+    #: Subnegotation sub-opion
+    subn_suboption = auto()
+    #: Subnegotation data
+    subn_data = auto()
+    #: Subnegotation end
+    subn_end = auto()
```
Comment (vfazio) [inline src/telnetio/_machine.py:41]: This is a bit nit-picky, but I prefer to see enums as uppercase. We're already following this convention for `ErrorKind`. This follows PEP-8 re constants: https://peps.python.org/pep-0008/#constants and follows Guido's recommendation: https://mail.python.org/pipermail/python-ideas/2016-September/042340.html

File: src/telnetio/_machine.py [hunk 2]
```diff
@@ -26,245 +22,263 @@ class TelnetOption:
...
+    msg: bytes
+
+    def as_bytes(self) -> bytes:
+        return self.msg
+
+
 @dataclass(frozen=True)
 class Command(Event):
-    cmd: Opt
+    cmd: int
```
Comment (vfazio) [inline src/telnetio/_machine.py:77]: Not something i expect you to address, but i do prefer this being typed to make it more clear that there are restrictions on the valid command values.

Comment (uSpike) [reply src/telnetio/_machine.py:77]: Yeah, it was that way, but then you'd get a `ValueError` or equivalent if you received a weird byte on the wire, which is not great.  I think it's better if the logic handles the cmd value regardless.

File: src/telnetio/_machine.py (hunk 1 above)
Comment (uSpike) [reply src/telnetio/_machine.py:41]: OK i can do that

Comment (uSpike) [reply src/telnetio/_machine.py]: fixed

**Comment by vfazio** [comment]
Lgtm

**Pull request merged**